schema/resolvers/tools.js

```diff
@@ -34,9 +27,40 @@ const toolsResolver = {
         },
       };
     },
-    tool: async (_, args, { toolLoader }, info) => {
-      console.log('ran tool');
-      const tool = await mongoDao.getOneDoc(database, 'tools', '_id', ObjectID(args._id));
+    getToolsByCategory: async (parent, { cursor, limit = 9, category }, context, info) => {
+      const cursorOptions = cursor
+        ? [
+            {
+              createdAt: {
+                $lt: fromCursorHash(cursor),
+              },
+            },
+          ]
+        : {};
+
+      const pipeline = [
+        {
+          $match: {
+            $and: [{ category: { $in: [category] } }],
+          },
+        },
+        { $sort: { createdAt: -1 } },
+        { $limit: limit + 1 },
+      ];
```

Comment:
put the pipeline and all other queries in to a designated folder just for that endpoint. e.g. for tools related, you can put all code into a tools folder, with it's own constant, its own query or helper folders. in this vase it's a lot easier to manage in the future.